### PR TITLE
Remove Retry timeout parameter from test helper

### LINKED ISSUE #20334: Test failure: CSharpCodeActions.GFUFuzzyMatchAfterRenameTracking: ArgumentOutOfRange in DateTime.AddTicks()
[Failure](https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_vs-integration_prtest/5490/testReport/junit/Roslyn.VisualStudio.IntegrationTests.CSharp/CSharpCodeActions/GFUFuzzyMatchAfterRenameTracking/) hit in PR #20277.

`WaitForLightBulbSession` calls `Retry` which calls  `beginTime.Add(TimeSpan.MaxValue)`:
```
MESSAGE:
System.ArgumentOutOfRangeException : The added or subtracted value results in an un-representable DateTime.\r\nParameter name: value
+++++++++++++++++++
STACK TRACE:
Server stack trace: at System.DateTime.AddTicks(Int64 value) at
System.DateTime.Add(TimeSpan value) at Microsoft.VisualStudio.IntegrationTest.Utilities.Helper.Retry[T](Func`1 action, TimeSpan delay, TimeSpan timeout) at
Microsoft.VisualStudio.IntegrationTest.Utilities.LightBulbHelper.WaitForLightBulbSession(ILightBulbBroker broker, IWpfTextView view) at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.<>c.<WaitForLightBulbSession>b__3_0(IWpfTextView view) at
Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.<>c__DisplayClass11_0.<GetExecuteOnActionViewCallback>b__0() at
System.Windows.Threading.DispatcherOperation.InvokeDelegateCore() at System.Windows.Threading.DispatcherOperation.InvokeImpl() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Windows.Threading.DispatcherOperation.Wait(TimeSpan timeout) at
System.Windows.Threading.Dispatcher.InvokeImpl(DispatcherOperation operation, CancellationToken cancellationToken, TimeSpan timeout) at
System.Windows.Threading.Dispatcher.Invoke(Action callback, DispatcherPriority priority, CancellationToken cancellationToken, TimeSpan timeout) at

System.Windows.Threading.Dispatcher.Invoke(Action callback) at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.InProcComponent.InvokeOnUIThread(Action action) at

Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.ExecuteOnActiveView(Action`1 action) at
Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.WaitForLightBulbSession() at
System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs) at
System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg) Exception rethrown at [0]: at
System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg) at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type) at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.WaitForLightBulbSession() at
Microsoft.VisualStudio.IntegrationTest.Utilities.OutOfProcess.TextViewWindow_OutOfProc.WaitForLightBulbSession() at
Microsoft.VisualStudio.IntegrationTest.Utilities.OutOfProcess.TextViewWindow_OutOfProc.InvokeCodeActionList() at
Roslyn.VisualStudio.IntegrationTests.CSharp.CSharpCodeActions.OrderFixesByCursorProximityLeft() in q:\roslyn\src\VisualStudio\IntegrationTest\IntegrationTests\CSharp\CSharpCodeActions.cs:line 270```

### PR DESCRIPTION
**Pull request by cston**
Fixes #20334.

### CODE REVIEW

**Comment by cston** [comment]
@dotnet-bot test windows_release_vs-integration_prtest please

**Comment by cston** [comment]
@dotnet-bot test windows_debug_vs-integration_prtest please

**Comment by cston** [comment]
@dotnet/roslyn-ide please review.

**Comment by cston** [comment]
@dotnet-bot test windows_debug_vs-integration_prtest please

**Comment by cston** [comment]
@dotnet/roslyn-ide please review.

File: src/VisualStudio/IntegrationTest/TestUtilities/Helper.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public static T Retry<T>(Func<T> action, TimeSpan delay, TimeSpan timeout)
                     System.Threading.Thread.Sleep(delay);
                 }
             }
-            while (beginTime.Add(timeout) > DateTime.UtcNow);
+            while (timeout == TimeSpan.MaxValue || beginTime.Add(timeout) > DateTime.UtcNow);
```
Comment (tannergooding) [inline src/VisualStudio/IntegrationTest/TestUtilities/Helper.cs]: I thought, in the initial design, we had decided to not allow any timeouts and to just retry indefinitely....

@jasonmalinowski, @DustinCampbell

Comment (jasonmalinowski) [inline src/VisualStudio/IntegrationTest/TestUtilities/Helper.cs]: Indeed, not sure why we have a timeout here. We should just delete the timeout parameter.

Comment (cston) [inline src/VisualStudio/IntegrationTest/TestUtilities/Helper.cs]: Removed.

**Comment by cston** [comment]
@dotnet-bot test windows_debug_vs-integration_prtest please

**Comment by tannergooding** [comment]
nit: Change the title of the PR so the resulting merge commit has an accurate message.

**Comment by cston** [comment]
@tannergooding, @jasonmalinowski, any additional concerns?

**Pull request merged**